src/canvas.cpp

```diff
@@ -598,6 +598,26 @@ void Canvas::copyToBitmap(int srcX, int srcY, Bitmap const * bitmap)
 }
 
 
+void Canvas::drawTransformedBitmap(Bitmap const * bitmap, float const transform[9])
```

Comment:
this function currently accepts a transform, and generates its inverse

we need the inverse to work out which are the source pixels to use when rendering a bitmap.  _generating_ an inverse matrix however is a _slightly_ costly operation - for repeated uses of the same transform, this is an operation that should not be strictly necessary

with that in mind, this function should probably include an optional inverse matrix argument, allowing for the calling code to pre-calculate the inverse